### PR TITLE
Release GVL around {,f}getattrlist calls in dir.c

### PR DESCRIPTION
**Pull request by jeremyevans**
Fixes [Bug #20587]

### CODE REVIEW

File: dir.c [hunk 1]
```diff
@@ -155,10 +182,17 @@ need_normalization(DIR *dirp, const char *path)
...
+    args.size = sizeof(attrbuf);
+    args.options = 0;
 #   if defined HAVE_FGETATTRLIST
-    int ret = fgetattrlist(dirfd(dirp), &al, attrbuf, sizeof(attrbuf), 0);
+    args.fd = dirfd(dirp);
+    int ret = IO_WITHOUT_GVL_INT(nogvl_fgetattrlist, &args);
 #   else
-    int ret = getattrlist(path, &al, attrbuf, sizeof(attrbuf), 0);
+    args.path = path;
+    int ret = IO_WITHOUT_GVL_INT(nogvl_getattrlist, &args);
```
Comment (nobu) [inline dir.c:195]: Rather than repeating similar calls, what about wrapper functions?

**Review by nobu — APPROVED** [review]

File: dir.c (hunk 1 above)
Comment (jeremyevans) [reply dir.c]: Pushed a commit to use wrapper functions.  Assuming it passes CI, I'll squash merge it.

**Comment by nobu** [comment]
My intention was including initializations: https://github.com/nobu/ruby/pull/new/getattrlist_args

**Comment by jeremyevans** [comment]
Sorry I misunderstood.  I agree your approach improves things.  I've submitted a pull request for it.

**Pull request merged**